src/msw/graphics.cpp

```diff
@@ -2730,6 +2731,12 @@ wxGraphicsContext * wxGDIPlusRenderer::CreateContext( const wxMemoryDC& dc)
 
     wxGDIPlusContext* context = new wxGDIPlusContext(this, dc);
     context->EnableOffset(true);
+
+    // GDI+ uses the DPI of the main display by default, so we don't need to do
+    // anything if the scale factor of the associated bitmap is already the
+    // same, but we do need to scale it correctly if it is different.
+    context->SetContentScaleFactor(dc.GetContentScaleFactor() /
```

Comment:
This doesn't work in the following case: system DPI 200%. Change DPI to 100% (don't log out/in). Save image with GDI+ renderer. (I assume the same will happen when moving from a 200% monitor to a 100% monitor).

I think we don't want the actual display DPI here / scale factor (now 1.0), but the system DPI / scale factor (the original 2.0). This works for me:
```context->SetContentScaleFactor(dc.GetContentScaleFactor() / (wxGetDPIofHDC(ScreenHDC()).x / wxDisplay::GetStdPPIValue()));```

Reply:
Instead of `ScreenHDC()` it also seems to work with `dc.GetHDC()`.

Reply:
Thanks, I didn't see it because when you move from one monitor to the other one the DPI of the main monitor doesn't change, but you're clearly right and we shouldn't use the current DPI here.

I'll use
```cpp
    const int defDPI = wxGetDPIofHDC(GetHdcOf(dc)).y;
    context->SetContentScaleFactor(dc.GetContentScaleFactor() * 96.0 / defDPI);
```
instead.